src/Instrumentation/EventWithDuration.php

```diff
@@ -0,0 +1,67 @@
+<?php
+/**
+ * Class EventWithDuration.
+ *
+ * @package AmpProject\AmpWP
+ */
+
+namespace AmpProject\AmpWP\Instrumentation;
+
+/**
+ * A server-timing event with a duration.
+ *
+ * @package AmpProject\AmpWP
+ */
+class EventWithDuration extends Event {
+
+	/**
+	 * Event duration.
+	 *
+	 * @var float
+	 */
+	protected $duration = 0.0;
+
+	/**
+	 * Event constructor.
+	 *
+	 * @param string        $name        Event name.
+	 * @param string|null   $description Optional. Event description.
+	 * @param string[]|null $properties  Optional. Additional properties for the event.
+	 * @param float         $duration    Optional. Event duration.
+	 */
+	public function __construct( $name, $description = null, $properties = [], $duration = 0.0 ) {
+		parent::__construct( $name, $description, $properties );
+		$this->duration = $duration;
+	}
+
+	/**
+	 * Set the event duration.
+	 *
+	 * @param float $duration Event duration.
+	 */
+	public function set_duration( $duration ) {
+		$this->duration = $duration;
+	}
+
+	/**
+	 * Get the event duration.
+	 *
+	 * @return float
+	 */
+	public function get_duration() {
+		return $this->duration;
+	}
+
+	/**
+	 * Get the server timing header string.
+	 *
+	 * @return string Server timing header string representing this event.
+	 */
+	public function get_header_string() {
+		return sprintf(
+			'%s;dur="%.1f"',
```

Comment:
Why only 1 decimal digit of precision? I suppose that's enough since `dur` is already in milliseconds?

Reply:
Yes, that's what seems to be recommended for use with server timings, as more digits increase header size without necessarily adding much in terms of analytics value. Elements that are so quick to process to fall below that threshold shouldn't probably be measured for server-timings anyway.